src/parser/sparqlParser/SparqlQleverVisitor.h

```diff
@@ -249,12 +249,42 @@ class SparqlQleverVisitor : public SparqlAutomaticVisitor {
 
   antlrcpp::Any visitOrderClause(
       SparqlAutomaticParser::OrderClauseContext* ctx) override {
-    return visitChildren(ctx);
+    vector<OrderKey> _orderBy;
+    for (auto* orderCondition : ctx->orderCondition()) {
+      _orderBy.push_back(visitOrderCondition(orderCondition).as<OrderKey>());
+    }
+    return _orderBy;
   }
 
   antlrcpp::Any visitOrderCondition(
       SparqlAutomaticParser::OrderConditionContext* ctx) override {
-    return visitChildren(ctx);
+    if (ctx->var()) {
+      return OrderKey{VariableOrderKey(ctx->var()->getText())};
+    } else if (ctx->constraint()) {
+      auto expr = sparqlExpression::SparqlExpressionPimpl{
+          std::move(visitConstraint(ctx->constraint()).as<ExpressionPtr>())};
+      bool desc = false;
+      bool exprIsVariable = expr.getVariableOrNullopt().has_value();
+      if (exprIsVariable) {
+        string var = expr.getVariableOrNullopt().value();
+        return OrderKey{VariableOrderKey(var, desc)};
+      } else {
+        return OrderKey{ExpressionOrderKey{std::move(expr), desc}};
+      }
+    } else if (ctx->brackettedExpression()) {
+      auto expr = sparqlExpression::SparqlExpressionPimpl{
+          std::move(visitBrackettedExpression(ctx->brackettedExpression())
+                        .as<ExpressionPtr>())};
+      bool desc = ctx->DESC() != nullptr;
+      auto exprIsVariable = expr.getVariableOrNullopt().has_value();
+      if (exprIsVariable) {
+        string var = expr.getVariableOrNullopt().value();
+        return OrderKey{VariableOrderKey(var, desc)};
+      } else {
+        return OrderKey{ExpressionOrderKey{std::move(expr), desc}};
+      }
```

Comment:
There is so so so much code duplication here, 
factor the two cases out into a lambda that probably takes
* The `any` (even the `as`-call is the same)
* The bool `isDescending`
and returns an `OrderKey` or maybe even bettern an `any` that contains an  OrderKey.